File: ASP Core - Backend/Dtos/User/LoginDTO.cs
using System.ComponentModel.DataAnnotations;

namespace aspcorebackend.Dtos.User {
    public class LoginDTO {
        [Required(ErrorMessage = "Username or email is required")]
        public string UsernameOrEmail { get; set; } = null!;

        [Required(ErrorMessage = "Password is required")]
        public string Password { get; set; } = null!;
    }
}

File: Spring Boot - Backend/src/main/java/com/booknav/springbackend/dto/User/LoginDTO.java
package com.booknav.springbackend.dto.User;

import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class LoginDTO {
    @NotBlank(message = "Username or Email is required")
    private String usernameOrEmail;
    @NotBlank(message = "Password is required")
    private String password;
}
